### PR TITLE
fix python urllib3 test

### PR DESCRIPTION
**Pull request by mandeepsplaha**
###### Merge Checklist  

**All** boxes should be checked before merging the PR *(just tick any boxes which don't apply to this PR)*
- [x] The toolchain has been rebuilt successfully (or no changes were made to it)
- [x] The toolchain/worker package manifests are up-to-date
- [x] Any updated packages successfully build (or no packages were changed)
- [x] Packages depending on static components modified in this PR (Golang, `*-static` subpackages, etc.) have had their `Release` tag incremented.
- [x] Package tests (%check section) have been verified with RUN_CHECK=y for existing SPEC files, or added to new SPEC files
- [x] All package sources are available
- [x] cgmanifest files are up-to-date and sorted (`./cgmanifest.json`, `./toolkit/scripts/toolchain/cgmanifest.json`, `.github/workflows/cgmanifest.json`)
- [x] LICENSE-MAP files are up-to-date (`./SPECS/LICENSES-AND-NOTICES/data/licenses.json`, `./SPECS/LICENSES-AND-NOTICES/LICENSES-MAP.md`, `./SPECS/LICENSES-AND-NOTICES/LICENSE-EXCEPTIONS.PHOTON`)
- [x] All source files have up-to-date hashes in the `*.signatures.json` files
- [x] `sudo make go-tidy-all` and `sudo make go-test-coverage` pass
- [x] Documentation has been updated to match any changes to the build system
- [x] Ready to merge

---

###### Summary 

What does the PR accomplish, why was it needed?
This PR fixes the failing test called `test_recent_date` by updating the source file containing the RECENT_DATE defined in file `src/urllib3/connection.py` which is used in the test.

###### Change Log  

Package: python-urllib3
Change: Fix test_recent_date test by updating the hard-coded date in file `src/urllib3/connection.py`.

###### Does this affect the toolchain?  

**NO**

###### Test Methodology

- Failing Test build: [479264](https://dev.azure.com/mariner-org/mariner/_build/results?buildId=479264&view=ms.vss-test-web.build-test-results-tab&runId=153941&resultId=100190&paneView=attachments)
- Passing Test build: Reusing the buddy build from main as 3.0-dev build is not functional - [485241](https://dev.azure.com/mariner-org/mariner/_build/results?buildId=485241&view=ms.vss-test-web.build-test-results-tab)

### CODE REVIEW

**Review by PawelWMS — APPROVED** [review]
Same ask as in #7294.

File: SPECS/python-urllib3/python-urllib3.spec [hunk 1]
```diff
@@ -1,14 +1,15 @@
...
-Release:        1%{?dist}
+Release:        2%{?dist}
 License:        MIT
 Vendor:         Microsoft Corporation
 Distribution:   Mariner
 Group:          Development/Languages/Python
 URL:            https://pypi.python.org/pypi/urllib3
 Source0:        https://github.com/urllib3/urllib3/archive/%{version}.tar.gz#/%{name}-%{version}.tar.gz
 BuildArch:      noarch
+Patch0: urllib3_test_recent_date.patch
```
Comment (PawelWMS) [inline SPECS/python-urllib3/python-urllib3.spec:12]: Please align spacing with the other tags above.

Comment (mandeepsplaha) [reply SPECS/python-urllib3/python-urllib3.spec:12]: Good call. Thanks.

**Comment by PawelWMS** [comment]
> Same ask as in #7294.

This is a tricky one, as upstream already makes this change in their `main` branch but haven't yet created a new release containing it. Since we haven't updated this package to the latest version, yet, I'm fine with taking this PR as-is and potentially removing it as we upgrade the whole package to the latest version (which may then contain the change from this patch).

**Pull request merged**